src/osp/Active/ActiveScene.cpp

```diff
@@ -258,6 +258,12 @@ void ActiveScene::draw(entt::entity camera)
     m_renderOrder.call(cameraComp);
 }
 
+void ActiveScene::system_machine_add(std::string const& name,
+                                     std::unique_ptr<ISysMachine>&& sysMachine)
+{
+    m_sysMachines.emplace(name, std::move(sysMachine));
+}
+
 MapSysMachine::iterator ActiveScene::system_machine_find(const std::string &name)

```

Comment:
If you change "m_sysMachines" to have the "std::less<>" comparitor (e.g. transparent comparison), then you can change this function to take an std::string_view instead of an std::string, and still get the same comparison behavior.